Scripts/Python/xLocTools.py

```diff
@@ -82,38 +82,41 @@ def MemberStatusString():
 
 xTranslatedAgeNames = {
     # City link-in points
-    "Ferry Terminal":       "Global.AgeNames.FerryTerminal",
-    "Tokotah Alley":        "Global.AgeNames.TokotahAlley",
-    "Palace Alcove":        "Global.AgeNames.PalaceAlcove",
-    "Library Courtyard":    "Global.AgeNames.LibraryCourtyard",
-    "Concert Hall Foyer":   "Global.AgeNames.ConcertHallFoyer",
+    "Ferry Terminal": PtGetLocalizedString("Global.AgeNames.FerryTerminal"),
+    "Tokotah Alley": PtGetLocalizedString("Global.AgeNames.TokotahAlley"),
+    "Palace Alcove": PtGetLocalizedString("Global.AgeNames.PalaceAlcove"),
+    "Library Courtyard": PtGetLocalizedString("Global.AgeNames.LibraryCourtyard"),
+    "Concert Hall Foyer": PtGetLocalizedString("Global.AgeNames.ConcertHallFoyer"),
     # Age names
-    "Eder Kemo":            "Global.AgeNames.EderKemo",
-    "Eder Gira":            "Global.AgeNames.EderGira",
-    "Gahreesen":            "Global.AgeNames.Gahreesen",
-    "Kadish":               "Global.AgeNames.Kadish",
-    "Nexus":                "Global.AgeNames.Nexus",
-    "Neighborhood":         "Global.AgeNames.Neighborhood",
-    "Relto":                "Global.AgeNames.Relto",
-    "Teledahn":             "Global.AgeNames.Teledahn",
+    "Eder Kemo": PtGetLocalizedString("Global.AgeNames.EderKemo"),
+    "Eder Gira": PtGetLocalizedString("Global.AgeNames.EderGira"),
+    "Gahreesen": PtGetLocalizedString("Global.AgeNames.Gahreesen"),
+    "Kadish": PtGetLocalizedString("Global.AgeNames.Kadish"),
+    "Nexus": PtGetLocalizedString("Global.AgeNames.Nexus"),
+    "Neighborhood": PtGetLocalizedString("Global.AgeNames.Neighborhood"),
+    "Relto": PtGetLocalizedString("Global.AgeNames.Relto"),
+    "Teledahn": PtGetLocalizedString("Global.AgeNames.Teledahn"),
     # Great Zero link-in points
-    "Great Zero Observation": "Global.AgeNames.GreatZeroObservation",
-    "Great Zero":           "Global.AgeNames.GreatZero",
+    "Great Zero Observation": PtGetLocalizedString("Global.AgeNames.GreatZeroObservation"),
+    "Great Zero": PtGetLocalizedString("Global.AgeNames.GreatZero"),
 }
 
+def CreatePossessiveString(possessorName, possessionName):
 
-def SafeEncode(text):
-    try:
-        #convert 8bit ascii to 16bit unicode characters 
-        encodedText = str(text)
-    except:          #exception called if the incomming string is already in unicode!
-        encodedText=text
-    return encodedText
+    if possessorName is None or possessionName is None:
+        return None
+
+    currLang = PtGetLanguage()
+    if possessorName[:-1] == "s" and currLang in {PtLanguage.kEnglish, PtLanguage.kGerman}:
```

Comment:
```suggestion
    if possessorName[:-1] == "s" and currLang in (PtLanguage.kEnglish, PtLanguage.kGerman):
```

Since these are integers, using a set for the membership test winds up being more expensive.

Reply:
Oh. Sad sad sad. Array `in` is still better than checking `and (currLang == PtLanguage.kEnglish or currLang == PtLanguage.kGerman)`, though, right? Presumably yes, otherwise you would have suggested it. 🤔

Reply:
I actually wasn't sure which would be faster, so behold:
```
Python 3.9.1 (tags/v3.9.1:1e5d33e, Dec  7 2020, 16:33:24) [MSC v.1928 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import timeit
>>> timeit.timeit("1 in (1, 2)")
0.023114799999998326
>>> timeit.timeit("1 == 1 or 1 == 2")
0.033624799999998345
>>> timeit.timeit("1 in (2, 1)")
0.048329700000003584
>>> timeit.timeit("1 == 2 or 1 == 1")
0.049628800000000695
>>> timeit.timeit("1 in {1, 2}")
0.03620010000000207
>>> timeit.timeit("1 in {2, 1}")
0.030450299999998265
```

I did not expect to see `set` outperform the `tuple` variant in its worst case configuration.

Reply:
I see the numbers. I think I understand what they mean. What I don't know is if this changes your suggestion...